service/src/main/java/uk/gov/hmcts/reform/fpl/service/DirectionHelperService.java

```diff
@@ -193,6 +202,85 @@ public void addDirectionsToCaseDetails(CaseDetails caseDetails,
         });
     }
 
+    @SuppressWarnings("unchecked")
+    public List<Element<Direction>> getDirectionsForUnrepresentedParties(CaseDetails caseDetails,
+                                                                         List<Element<Direction>> directions) {
+        final boolean direction = true;
+
+        return directions.stream()
+            .filter(element -> {
+                    List<Element<UUID>> ids = new ArrayList<>();
+
+                    List<Element<Representative>> representatives = defaultIfNull(
+                        (List<Element<Representative>>) caseDetails.getData().get("representatives"), emptyList());
+
+                    // if assigned to specific person and representatives exist -> find representative ids
+                    // if not -> return direction.
+                    if (directionIsValidForFiltering(element, representatives)) {
+                        if (element.getValue().getParentsAndRespondentsAssignee() != null) {
+                            int assigneeValue = element.getValue().getParentsAndRespondentsAssignee().ordinal();
+
+                            List<Element<Respondent>> respondents =
+                                (List<Element<Respondent>>) caseDetails.getData().get("respondents1");
+
+                            // if respondent exist -> set ids.
+                            // else return direction
+                            if (assigneeValue < respondents.size()) {
+                                ids = respondents.get(assigneeValue).getValue().getRepresentedBy();
+                            } else {
+                                return direction;
+                            }
+                        }
+
+                        if (element.getValue().getOtherPartiesAssignee() != null) {
+                            int assigneeValue = element.getValue().getOtherPartiesAssignee().ordinal();
+
+                            Others others = (Others) caseDetails.getData().get("others");
+
+                            List<Element<Other>> listOthers =
+                                defaultIfNull(others.getAdditionalOthers(), new ArrayList<>());
+
+                            listOthers.add(0, ElementUtils.element(others.getFirstOther()));
+
+                            if (assigneeValue < listOthers.size()) {
+                                ids = listOthers.get(assigneeValue).getValue().getRepresentedBy();
+                            } else {
+                                return direction;
+                            }
+                        }
+
+                    } else {
+                        // include direction in filter
+                        return direction;
+                    }
+
+                    List<Element<UUID>> finalIds = ids;
+
+                    if (ids.isEmpty()) {
+                        //
+                        return direction;
```

Comment:
Direction seems as something tangible, but this is boolean value, would you consider to rename?

Reply:
so I wanted this to make a bit more sense rather than returning true false. Can you think of a better name, I think we are getting at the same point. My thinking was, when true -> it adds the direction to the filtered list, hence return direction...

Reply:
So personally how I would implement is:
1. I would extract the lambda to separate method, calling it something like `isDirectionForUnrepresentedParty` -  that reduces cognitive load on the main method which becomes simple: get direction, stream, filter collect and removes direction field from that level.
2. If someone wants to see how filtering is done then can jump to this method. I think in the method itself returning true wouldn't hurt as we already are in the mindset of checking if the direction is for appropriate party. 

If you want to keep it as one bigger methood, probably I would suggest something: `directionForUnrepresentedParty`? Would it be better?